scale/job/configuration/interface/job_interface.py

```diff
@@ -280,3 +282,42 @@ def populate_command_argument_settings(self, command_arguments, job_configuratio
                     command_arguments = self._replace_command_parameter(command_arguments, setting_name, '')
 
         return command_arguments
+
+    def populate_command_argument_env_vars(self, command_arguments):
+        """Return the command arguments string, populated with the environment
+        variables.
+
+        :param command_arguments: The command_arguments that you want to perform the replacement on
+        :type command_arguments: string
+        :return: command arguments with the env_vars populated
+        :rtype: str
+        """
+
+        for env_var in self.definition['env_vars']:
+            env_var_name = env_var['name']
+            env_var_value = env_var['value']
+            command_arguments = self._replace_command_parameter(command_arguments,
+                                                                env_var_name,
+                                                                env_var_value)
+
+        return command_arguments
```

Comment:
The env vars are supposed to be substituted into the command arguments. The same substitution done into the command arguments is supposed to be done into the env vars.

So a command argument can have input params, settings, and the job output dir replaced in its string, e.g. ${param1} ${setting1} ${job_output_dir}.

Env vars may have the same substitution performed with their values e.g. "env_vars": [{"name": "var1", "value": "${param1}"].

Also the env vars need to be passed from the job interface into the Docker params in the job_exe model's job_configuration. So at schedule time, pull env vars from interface, perform any substitution on their values, and insert them into the job_configuration Docker params.